File: typescript/Structural/07-Bridge/Notification/NormalNotification.ts
import { INotificationChannel } from "../Channel/INotificationChannel";
import { JobNotification } from "./JobNotification";

export class NormalNotification extends JobNotification {
  constructor(channel: INotificationChannel) {
    super(channel);
  }

  notify(jobId: string, recipient: string, message: string): void {
    this.channel.send(recipient, `Job #${jobId} Update`, message);
  }
}

File: csharp/Structural/07-Bridge/Notification/NormalNotification.cs
using Bridge.Channel;

namespace Bridge.Notification;

public class NormalNotification : JobNotification
{
    public NormalNotification(INotificationChannel channel) : base(channel) { }

    public override void Notify(string jobId, string recipient, string message)
    {
        Channel.Send(recipient, $"Job #{jobId} Update", message);
    }
}
